find_dapper_test.go

```diff
@@ -0,0 +1,399 @@
+		)
+	})
+
+	t.Run("If a user holds an invalid find name, get status should not return it", func(t *testing.T) {
+
+		nameAddress := otu.O.Address("user2")
+		otu.moveNameTo("user2", "user1", "user2")
+		otu.O.Script("getStatus",
+			WithArg("user", nameAddress),
+		).AssertWithPointerError(t,
+			"/FINDReport/profile/findName",
+			"Object has no key 'findName'",
+		)
+	})
+
+	t.Run("Should be able to create and edit the social link", func(t *testing.T) {
+
+		otu.O.Tx("editProfileDapper",
+			WithSigner("user1"),
+			WithArg("name", "user1"),
+			WithArg("description", "This is description"),
+			WithArg("avatar", "This is avatar"),
+			WithArg("tags", `["This is tag"]`),
+			WithArg("allowStoringFollowers", true),
+			WithArg("linkTitles", map[string]string{"CryptoTwitter": "0xBjartek", "FindTwitter": "find"}),
+			WithArg("linkTypes", map[string]string{"CryptoTwitter": "Twitter", "FindTwitter": "Twitter"}),
+			WithArg("linkUrls", map[string]string{"CryptoTwitter": "https://twitter.com/0xBjartek", "FindTwitter": "https://twitter.com/findonflow"}),
+			WithArg("removeLinks", "[]"),
+		).
+			AssertSuccess(t)
+
+		otu.O.Script("getStatus",
+			WithArg("user", "user1"),
+		).AssertWithPointerWant(t,
+			"/FINDReport/profile/links/FindTwitter",
+			autogold.Want("getStatus Find twitter", map[string]interface{}{
+				"title": "find",
+				"type":  "Twitter",
+				"url":   "https://twitter.com/findonflow",
+			}),
+		)
+
+		otu.O.Tx("editProfileDapper",
+			WithSigner("user1"),
+			WithArg("name", "user1"),
+			WithArg("description", "This is description"),
+			WithArg("avatar", "This is avatar"),
+			WithArg("tags", `["This is tag"]`),
+			WithArg("allowStoringFollowers", true),
+			WithArg("linkTitles", "{}"),
+			WithArg("linkTypes", "{}"),
+			WithArg("linkUrls", "{}"),
+			WithArg("removeLinks", `["FindTwitter"]`),
+		).
+			AssertSuccess(t)
+
+		otu.O.Script("getStatus",
+			WithArg("user", "user1"),
+		).AssertWithPointerError(t,
+			"/FINDReport/profile/links/FindTwitter",
+			"Object has no key 'FindTwitter'",
+		)
+
+	})
+
+	t.Run("Should be able to buy addons that are on Network", func(t *testing.T) {
+
+		user := "user1"
+
+		otu.buyForgeDapper(user)
+
+		/* Should not be able to buy addons with wrong balance */
+		otu.O.Tx("buyAddonDapper",
+			WithSigner("user1"),
+			WithPayloadSigner("account"),
+			WithArg("merchAccount", "find"),
+			WithArg("name", "name1"),
+			WithArg("addon", "forge"),
+			WithArg("amount", 10.0),
+		).
+			AssertFailure(t, "Expect 50.00000000 FUSD for forge addon")
+
+		/* Should not be able to buy addons that does not exist */
+		otu.O.Tx("buyAddonDapper",
+			WithSigner(user),
+			WithPayloadSigner("account"),
+			WithArg("merchAccount", "find"),
+			WithArg("name", user),
+			WithArg("addon", "A.f8d6e0586b0a20c7.Dandy.NFT"),
+			WithArg("amount", 10.0),
+		).
+			AssertFailure(t, "This addon is not available.")
+
+	})
+
+	// Should not be able to manage fund
```

Comment:
can we just remove this if it is not relevant anymore?

Reply:
removed